### DISCUSSION TITLE
English Language Resources for SimpleSDXL

### DISCUSSION THREAD

**Discussion by DavidDragonsage** [metercai/SimpleSDXL#68] [General]
There are currently two English language videos that discuss the installation and basic use of SimpleSDXL:
https://www.youtube.com/watch?v=qJxCLIF2rJc
and
https://www.youtube.com/watch?v=IREkiuRpQiY
I am sure there will be more videos coming!

MeterCai posted a short introduction to SimpleSDXL here, which is how I found out about it:
https://github.com/lllyasviel/Fooocus/discussions/3548

And our Facebook Pure Fooocus group has a lot of discussion and information on SimpleSDXL, with some exclusive videos and sample images. We also have a Guide devoted to SimpleSDXL which is gradually growing as we learn more about it:
https://www.facebook.com/groups/fooocus

I am very excited by this new development in the Fooocus world and really appreciate all the work that has gone into the SimpleSDXL project. Thank you! ❤️

**Comment by metercai** [comment]
thanks. If there are any English expressions on the interface that do not conform to common usage, you can revise them in en. json, send them to me, and update them to the version package.

**Reply by DavidDragonsage** [reply]
Thank you, I appreciate that. ❤️

**Comment by metercai** [comment]
@DavidDragonsage Have you ever revised an updated version of en.json that is suitable for English users

**Comment by DavidDragonsage** [comment]
@metercai I have not edited this particular json but it looks pretty easy. None of the English stands out to me as being non-idiomatic. But if you would like something changed, just let me know.

But related to language usage, I do think that there are some problems with the "Additional Prompt Quick List" that is displayed when the user selects Inpaint: Improve Detail method.

Currently there is the phrase "detailed girl face" where the intent is really "detailed woman's face". Using "girl" to describe an adult woman is not only disrespectful but I did a study on this at Pure Fooocus that showed that this prompt usage confuses SDXL base models.

When used as the main prompt, "girl" can be interpreted as "little girl", "teenage girl", "young woman" or "adult woman". When used to Inpaint an adult woman's face, the result will tend to be somewhat "girl-like". In particular, the eyes will be tend to be larger than they should be, out of scale with the face.

I heard that Flux is much more particular about this sloppy usage of the word "girl". If you put "girl" in the prompt it will likely generate a female child. I have not actually tried this.

Another non-idiomatic usage in this "Additional Prompt Quick List" is "detailed man face" instead of "detailed man's face". In this case the prompt processor will give identical results, but this usage of bad grammar is unprofessional.

This list was inherited from mainline Fooocus. Of course the user can reconfigure the list in config.sys, but many users are not comfortable doing so. I do not think that users should have to - the default should already present a good example.

**Reply by DavidDragonsage** [reply]
Yes, I just checked with Flux1 Dev. When I use "girl" in the main prompt the image is always a "little girl", never a woman. I appreciate this precision! 🙂

**Reply by DavidDragonsage** [reply]
If you are asking about European English versus the American English that Fooocus was written in, the biggest difference is the use of "colour" instead of "color" in en.json

The styles were made by several different authors and the capitalization of acronyms is inconsistent. For example, we have "3D" but also "2d". All acronyms should be capitalized, such as "GTA", "HDR", "DMT Art Style", "RPG".

There is a known mistake in the styles list that has never been corrected. "Mandola Art" should be "Mandala Art".

**Comment by metercai** [comment]
There may be a misunderstanding, but I am referring to whether the English expression on the newly added UI is appropriate. If improvement is needed, it can be optimized and improved using the en.json .
Recently, a friend provided a French version of fr.json.

**Comment by DavidDragonsage** [comment]
Well you got a bonus - *all* the problems in English that I could think of, including those in en.json!

Just point me in the right direction and I can correct or improve all these issues for you.

There another small language issue. Under the Setting tab there is a block of untranslated text, as shown in the attachment. This is the text referring to the default model, JuggernautXL XI.

Incidentally, I agree that this model is a good default, although its limited depth of field does not make it a good choice for landscapes - there are better general purpose models. However, beginners are typically focused on making portraits so this version of Juggernaut is an excellent choice for the default, and currently the best in the Juggernaut series. There is also the Fooocus history of supporting Juggernaut as the default.

![Juggernaut XI Message](https://github.com/user-attachments/assets/d42a42b8-84d5-45c8-a183-7513c55f7680)

**Comment by DavidDragonsage** [comment]
There is an awkwardness in "Finished Images Catalog" that confused me when I first started using SimpleSDXL2. These are not necessarily "finished images", often they need further work before upload. Under the Enhanced tab, this feature is referred to as "historical gallery images" (see attachment).

I think a better description for "Finished Images Catalog" would be "Image Gallery". "Gallery" creates a very positive and artistic impression in English, but a "catalog" is simply something that you shop from. Also "catalog" can create an annoyance for European English users because the word is spelled "catalogue". The word "gallery" avoids this.

![Historical Gallery Images](https://github.com/user-attachments/assets/49867d85-f689-4e57-abae-ba251e8742ad)

**Reply by DavidDragonsage** [reply]
Under the setting tab,
"Negative Prompt
Describing what you do not want to see."

"Describing" is in the wrong tense. This should be in the imperative or directive tense, like the rest of the UI. The phrase should be: "Describe what you do not want to see."

Under the Enhanced tab:
"Language of UI" and "Theme of background" are unnecessarily wordy for a program interface. They would be better as "UI Language" and "Background Theme".

The Fooocus interface has always been inconsistent in the use of capitalization for the titles of settings. In general, titles are fully capitalized, which is more appropriate - but there are many anomalies. For example, under the Enhanced tab, "Translation methods" would be better as "Translation Methods".

Consistent capitalization helps people focus on the features and settings that are available. I know that other languages such as German have quite different rules on capitalization - German tends to use more capitals than English - so capitalization if very language dependent.

**Comment by metercai** [comment]
The newly added UI expression is indeed unprofessional. Don't list them here, just revise your opinions into the en.json file

**Comment by DavidDragonsage** [comment]
Thank you, I appreciate that! I will be an honour to do so. You have done a great job. I really love this UI and if I can help in a small way to improve it I would enjoy doing so. ❤️ 

I will send you a separate text file as well so that you know what I have changed.

I actually do not know how to send it to you directly. Do I just post it as an attachment here?

**Reply by metercai** [reply]
yeah, attach it here. better, if you know how to use Git, you can submit pull requests.

**Reply by DavidDragonsage** [reply]
Yes, I did submit pull requests for mainline Fooocus. While I appreciate the controlled nature of the pull request system, I think that in this case the formality of a pull request is an unnecessary complication.

I have attached a revised en.json, as well as introducing en_uk.json for European English. Each file is also supplied with a change list.

Unfortunately there are several texts that are outside the scope of the language files, such as "Finished Images Catalog", but this is a good start.

[en.json](https://github.com/user-attachments/files/17158160/en.json)

[en_uk.json](https://github.com/user-attachments/files/17158162/en_uk.json)

[Changes to en.json.txt](https://github.com/user-attachments/files/17158164/Changes.to.en.json.txt)

[Additional Changes for en_uk.txt](https://github.com/user-attachments/files/17157758/Additional.Changes.for.en_uk.txt)

**Comment by DavidDragonsage** [comment]
This corrects an annoying mistake in sdxl_styles_diva.json.

This style lists refers to "Mandola Art" and includes erroneous keywords such as "mandola" within the style itself. A "mandala" refers to circle or square of religious symbols (https://en.wikipedia.org/wiki/Mandala) but a "mandola" is a musical instrument (https://en.wikipedia.org/wiki/Mandola). All references to "mandola" are changed to "mandala".

Needless to say, this style has never worked properly!

I have included the revised json file and also the standardized "cat" image for the samples folder. Included in a zip to protect the integrity of the jpg file.

[sdxl_styles_diva.zip](https://github.com/user-attachments/files/17158039/sdxl_styles_diva.zip)

**Comment by aleph23** [comment]
When using a face editor extension in A1111/SDnext/Forge/Comfy I have always found that even the simplest of SDv1.5 text encoders are familiar with the photographers' term 'headshot'. I have always had perfect performance using 'Professional headshot of a X-quality gendered-person doing/looking at Y.' 

i.e. if I want to get fancy, I might use: "Professional headshot of a striking 24-year-old woman with short dark hair and hazel eyes, squinting defiantly at the viewer." But--using the 'SD WebUI Face Editor' extension--I usually get good enough results just leaving a default, "Professional hdsht of a {gender}" and let the model infer the rest from what's already in the image.

Or, through the Yandex obfuscator: 当使用的脸编辑延伸 A1111/SDnext/Comfy/Forge 我们始终发现，即使是最简单的SDv1.5的文本编码器熟悉的摄影师'任期'headshot'. 我一直有完美的表现采用专业爆头的一个X质性别人做/寻找Y.' 我一直有完美的表现采用专业 'headshot' 的X质性别人做/寻找Y.'
